### PR TITLE
Add tracing of Hdfs class based on AbstractFileSystem

### PR DESCRIPTION
**Pull request by ashangit**
Some of our client are not only using FileSystem base classes
but also AbstractFileSystem though FileContext

### CODE REVIEW

File: agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java [hunk 1]
```diff
@@ -169,12 +169,13 @@ public static Object intercept(
...
         protected ElementMatcher<? super TypeDescription> typeMatcher() {
-            return nameStartsWith("org.apache.hadoop.hdfs.DistributedFileSystem");
+            return nameStartsWith("org.apache.hadoop.hdfs.DistributedFileSystem").or(nameStartsWith("org.apache.hadoop.fs.Hdfs"));
         }
 
         @Override
         protected ElementMatcher<? super MethodDescription> methodMatcher() {
-            return named("create").and(takesArguments(7)).and(takesArgument(0, pathTD));
+            return named("create").and(takesArguments(7)).and(takesArgument(0, pathTD))
+                .or(named("createInternal"));
```
Comment (mchataigner) [inline agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java]: no match on number of parameters?

File: agent/src/test/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracerTest.java [hunk 2]
```diff
@@ -180,47 +187,99 @@ private void checkEvent(String action, Path path, DataAccessEventProtos.FsEvent.
...
+            Options.ChecksumOpt.createDisabled(), false);
+
+        Class<?> clazzHdfsDataOutputStream = classLoader.loadClass(HdfsDataOutputStream.class.getName());
+        Method write = clazzHdfsDataOutputStream.getMethod("write", byte[].class);
+        Method close = clazzHdfsDataOutputStream.getMethod("close", null);
+        write.invoke(os, "This is a test".getBytes());
+        close.invoke(os);
+    }
+
+    private void createSrcPatDistributedFilesystem(Path file) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, ClassNotFoundException, IOException {
```
Comment (mchataigner) [inline agent/src/test/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracerTest.java]: typo missing h to `createSrcPatDistributedFilesystem`

File: agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java (hunk 1 above)
Comment (ashangit) [reply agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java]: only one createInternal method exist on Hdfs class so I don't put any other filtering here

File: agent/src/test/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracerTest.java (hunk 2 above)
Comment (ashangit) [reply agent/src/test/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracerTest.java]: in fact will do

**Pull request merged**